### PR TITLE
fix(c): improve doxygen comment matching

### LINKED ISSUE #6276: `///` comments are not parsed as doxygen
### Describe the bug

In C files,`///` comments are not parsed as doxygen

### To Reproduce

```c
/// @brief sdasdf
///
/// @param argc asdfasdf
/// @param argv asdfasdf
/// @return asdfsafd
int main(int argc, char ** argv)
{
    return 0;
}
```

### Expected behavior

Comment is parsed as doxygen

### Output of `:checkhealth nvim-treesitter`

```markdown
==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.21.0 (1c55abb5308fe3891da545662e5df7ba28ade275) (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v21.6.2 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: cc (GCC) 13.2.1 20230801
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "6.7.4-arch1-1",
  sysname = "Linux",
  version = "#1 SMP PREEMPT_DYNAMIC Mon, 05 Feb 2024 22:07:49 +0000"
} ~

Parser/Features         H L F I J
  - arduino             ✓ ✓ ✓ ✓ ✓
  - bash                ✓ ✓ ✓ . ✓
  - bibtex              ✓ . ✓ ✓ .
  - c                   ✓ ✓ ✓ ✓ ✓
  - cmake               ✓ . ✓ ✓ .
  - comment             ✓ . . . .
  - cpp                 ✓ ✓ ✓ ✓ ✓
  - css                 ✓ . ✓ ✓ ✓
  - csv                 ✓ . . . .
  - diff                ✓ . . . .
  - dockerfile          ✓ . . . ✓
  - dot                 ✓ . . . ✓
  - doxygen             ✓ . . ✓ ✓
  - git_config          ✓ . ✓ . ✓
  - git_rebase          ✓ . . . ✓
  - gitattributes       ✓ ✓ . . ✓
  - gitcommit           ✓ . . . ✓
  - gitignore           ✓ . . . .
  - gpg                 ✓ . . . ✓
  - graphql             ✓ . . ✓ ✓
  - hcl                 ✓ . ✓ ✓ ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - http                ✓ . . . ✓
  - ini                 ✓ . ✓ . .
  - java                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - jq                  ✓ ✓ . . ✓
  - jsdoc               x . . . .
  - json                ✓ ✓ ✓ ✓ .
  - jsonc               ✓ ✓ ✓ ✓ ✓
  - latex               ✓ . ✓ . ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - make                ✓ . ✓ . ✓
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - matlab              ✓ ✓ ✓ ✓ ✓
  - ninja               ✓ . ✓ ✓ .
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - rasi                ✓ ✓ ✓ ✓ .
  - regex               ✓ . . . .
  - rst                 ✓ ✓ . . ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - sql                 ✓ . . ✓ ✓
  - ssh_config          ✓ ✓ ✓ ✓ ✓
  - toml                ✓ ✓ ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓
  - xml                 ✓ ✓ ✓ ✓ ✓
  - yaml                ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

The following errors have been detected: ~
- ERROR jsdoc(highlights): Failed to load parser for language 'jsdoc': uv_dlopen: /home/chad/.local/share/nvim/lazy/nvim-treesitter/parser/jsdoc.so: undefined symbol: tree_sitter_jsdoc_external_scanner_create
  jsdoc(highlights) is concatenated from the following files:
  | [ERROR]:"/home/chad/.local/share/nvim/lazy/nvim-treesitter/queries/jsdoc/highlights.scm", failed to load: Failed to load parser for language 'jsdoc': uv_dlopen: /home/chad/.local/share/nvim/lazy/nvim-treesitter/parser/jsdoc.so: undefined symbol: tree_sitter_jsdoc_external_scanner_create
```
```


### Output of `nvim --version`

```text
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1702233742

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
```


### Additional context

_No response_

### PR DESCRIPTION
**Pull request by thetic**
Add support for `///` and `//!` comments.

Closes #6276

### CODE REVIEW

File: queries/c/injections.scm [hunk 1]
```diff
@@ -9,7 +9,7 @@
   (#set! injection.language "re2c"))
 
 ((comment) @injection.content
-  (#lua-match? @injection.content "/[*][!<*][^a-zA-Z]")
+  (#lua-match? @injection.content "/[*\/][!*\/][^a-zA-Z]")
```
Comment (ObserverOfTime) [inline queries/c/injections.scm:12]:
```suggestion
  (#lua-match? @injection.content "/[*\/][!<*\/][^a-zA-Z]")
```

File: queries/cpp/injections.scm [hunk 2]
```diff
@@ -5,7 +5,7 @@
   (#set! injection.language "comment"))
 
 ((comment) @injection.content
-  (#lua-match? @injection.content "/[*][!<*][^a-zA-Z]")
+  (#lua-match? @injection.content "/[*\/][!*\/][^a-zA-Z]")
```
Comment (ObserverOfTime) [inline queries/cpp/injections.scm:8]:
```suggestion
  (#lua-match? @injection.content "/[*\/][!<*\/][^a-zA-Z]")
```

File: queries/c/injections.scm (hunk 1 above)
Comment (thetic) [reply queries/c/injections.scm:12]: `//<` is not a doxygen comment.

Comment (ObserverOfTime) [reply queries/c/injections.scm:12]: https://www.doxygen.nl/manual/docblocks.html#memberdoc

Comment (thetic) [reply queries/c/injections.scm:12]:
```suggestion
  (#lua-match? @injection.content "/[*\/][!*\/]<?[^a-zA-Z]")
```

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**